### PR TITLE
Closes #1986 - File I/O Documentation Updates

### LINKED ISSUE #1986: `FILEIO.md` Updates
This should be completed after PR #1985 is merged.

FILEIO.md is outdated after the updates to our IO functionality to remove the virtual layer and have dedicated HDF5 and Parquet functions.

**Recommended Updates**
- Add a `file_io` folder to `pydoc`
- Add a `fileio_menu.rst` with a `toctree` to `pydoc/file_io`. Provide a general IO description and supported file types.
- Create `HDF5.md` and `Parquet.md` detailing interactions with each file types. Be sure to include the structure of data within the files. These files should be linked in the `toctree`
- Remove the current `FILEIO.md` file

### PR DESCRIPTION
**Pull request by Ethan-DeBandi99**
Closes #1986 

Updates the layout of the File I/O documentation to create a main page, a page for HDf5, a page for Parquet and a page for Import/Export. Also, updates docstrings for IO functionality to ensure that links in the File I/O section to API References have the necessary information. 

This update replaces the single `FILEIO.md`, so it has been removed.

For now, I included links to deprecated functionality since it is unclear when we will be actually removing these elements. My intention was to provide an easier to navigate File I/O section without overcrowding with duplicated information from the API reference. For this reason, I simple linked the necessary functions in from the API reference instead of regenerating that documentation. 

I maintained the `IO.rst` file, but unlinked it from the index for the time being. I can remove it all together, but was unsure if any information was really still relevant or not. @pierce314159, @joshmarshall1, @jaketrookman, @hokiegeek2  - please let me know if you believe any of that information needs to be moved into the File I/O section or maintained in another way. 

To review this fully, please pull down the PR and run `make doc`. You can then open `docs/index.html`  in your browser and navigate the IO section.

### CODE REVIEW

File: pydoc/file_io/HDF5.md [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+More information on HDF5 is available [here](https://www.hdfgroup.org/solutions/hdf5/).
+
+## File Configuration
+
+Arkouda supports saving HDF5 files in 2 ways:
+
+- Single File
+  - All data is pulled local to the processing root node and saved into one file
+- Distributed Files (Default)
+  - Each file contains the portion of the data local to the locale where the file is being written are saved. This results in one file per locale.
```
Comment (pierce314159) [inline pydoc/file_io/HDF5.md:14]: maybe reword `where the file is being written are saved`

File: pydoc/file_io/HDF5.md [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+**Truncate**
+> When writing to HDF5 in `truncate` mode, any existing HDF5 file with the same name will be overwritten. If no file exists, one will be created. If writing multiple objects, the first is written in `truncate` mode. All subsequent objects will then be appended to the file. The user will be notified of any overwritten files.
+
+**Append**
+> When writting to HDF5 in `append` mode, all datasets will be appended to the file. If no file with the supplied name exists, one will be created. If any datasets being written have a name that is already the name of a dataset within the file, an error will be generated.
+
+## Data Distribution
+
+**Single File**
+> If the user elects to write to a single HDF5 file, all data will be pulled to the processing node and saved to ONE file with the supplied file name. It is import to ensure that the object is small enough to prevent memory exhaustion on the node.
```
Comment (pierce314159) [inline pydoc/file_io/HDF5.md:130]: `import` -> `important`

File: pydoc/file_io/PARQUET.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+Data can also be saved using no compression. Arkouda now supports writting Parquet files with all compression types supported by Parquet.
+
+## Supported Write Modes
+
+**Truncate**
+> When writing to HDF5 in `truncate` mode, any existing HDF5 file with the same name will be overwritten. If no file exists, one will be created. If writing multiple objects, all corresponding columns will be written to the Paruqet file at once. 
+
+**Append**
+> When writting to HDF5 in `append` mode, all datasets will be appended to the file. If no file with the supplied name exists, one will be created. If any datasets being written have a name that is already the name of a dataset within the file, an error will be generated.
```
Comment (pierce314159) [inline pydoc/file_io/PARQUET.md:34]: both of these make references to `HDF5`

**Review by pierce314159 — COMMENTED** [review]
A few small things. I didn't give the most thorough review, but I think it's worth having a few typos make it in if it means having this in the next release

Overall restructuring looks good!

File: pydoc/file_io/HDF5.md [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+
+*Please Note - The offsets dataset is note required but can be provided. Strings uses null byte termination and is able to calculate the offsets of its components during reads.*
+
+## Supported Write Modes
+
+**Truncate**
+> When writing to HDF5 in `truncate` mode, any existing HDF5 file with the same name will be overwritten. If no file exists, one will be created. If writing multiple objects, the first is written in `truncate` mode. All subsequent objects will then be appended to the file. The user will be notified of any overwritten files.
+
+**Append**
+> When writting to HDF5 in `append` mode, all datasets will be appended to the file. If no file with the supplied name exists, one will be created. If any datasets being written have a name that is already the name of a dataset within the file, an error will be generated.
```
[the remark this answers is not in the document]
Comment (Ethan-DeBandi99) [reply pydoc/file_io/HDF5.md:125]: fixed

File: pydoc/file_io/HDF5.md (hunk 1 above)
Comment (Ethan-DeBandi99) [reply pydoc/file_io/HDF5.md:14]: Done

File: pydoc/file_io/HDF5.md (hunk 2 above)
Comment (Ethan-DeBandi99) [reply pydoc/file_io/HDF5.md:130]: done

File: pydoc/file_io/PARQUET.md (hunk 3 above)
Comment (Ethan-DeBandi99) [reply pydoc/file_io/PARQUET.md:34]: Updated to Parquet

File: arkouda/dataframe.py [hunk 5]
```diff
@@ -1475,12 +1475,21 @@ def to_hdf(self, path, index=False, columns=None, file_type="distribute"):
...
+        RuntimeError
+            Raised if a server-side error is thrown saving the pdarray
         Notes
         -----
         This method saves one file per locale of the arkouda server. All
         files are prefixed by the path argument and suffixed by their
         locale number.
+        See Also
+        ---------
+        to_hdf, load
```
Comment (joshmarshall1) [inline arkouda/dataframe.py:1492]: This `See Also` section should probably read `to_parquet, load`

**Review by joshmarshall1 — APPROVED** [review]
One small comment mistake, but nothing I'd hold up the merge over

File: arkouda/dataframe.py (hunk 5 above)
Comment (Ethan-DeBandi99) [reply arkouda/dataframe.py]: Updated

**Review by pierce314159 — APPROVED** [review]
